tools/idf.py

```diff
@@ -713,7 +713,32 @@ def main() -> None:
         else:
             raise
     else:
-        cli(sys.argv[1:], prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+        argv = expand_file_arguments(argv or sys.argv[1:])
+
+        cli(argv, prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+
+
+def expand_file_arguments(argv):
+    """
+    Any argument starting with "@" gets replaced with all values read from a text file.
+    Text file arguments can be split by newline or by space.
+    Values are added "as-is", as if they were specified in this order
+    on the command line.
+    """
+    new_args = []
+    expanded = False
```

Comment:
I think this flag is unnecessary. `if expanded` could be replaced by `if len(new_args) > 0`.

Reply:
In python, it's sufficient to check `if new_args`, no need to use len

Reply:
`new_args` also contains the non-file arguments in `argv`, so the flag is needed to detect if there were any expansions - but could probably rename `new_args` as it implies it only contains the newly expanded args.